File: 2026/04-April/08-Wednesday/solution.java
class Solution {
    public int xorAfterQueries(int[] nums, int[][] queries) {
        long MOD = 1000000007;

        for (int[] q : queries) {
            int l = q[0];
            int r = q[1];
            int k = q[2];
            int v = q[3];

            for (int i = l; i <= r; i += k) {
                nums[i] = (int)((nums[i] * 1L * v) % MOD);
            }
        }

        int result = 0;
        for (int num : nums) {
            result ^= num;
        }

        return result;
    }
}

File: 2026/04-April/08-Wednesday/solution.py
class Solution:
    def xorAfterQueries(self, nums, queries):
        MOD = 10**9 + 7

        for l, r, k, v in queries:
            i = l
            while i <= r: 
                nums[i] = (nums[i] * v) % MOD
                i += k

        result = 0
        for num in nums:
            result ^= num

        return result
